src/thunderbots/software/gui/visualizer_wrapper.cpp

```diff
@@ -0,0 +1,88 @@
+#include "software/gui/visualizer_wrapper.h"
+#include "software/gui/drawing/world.h"
+
+VisualizerWrapper::VisualizerWrapper(int argc, char** argv) :
+            ThreadedObserver<World>(),
+            ThreadedObserver<WorldDrawFunction>(),
+            ThreadedObserver<AIDrawFunction>(),
+            ThreadedObserver<PlayInfo>()
+{
+    auto application_promise =
+        std::make_shared<std::promise<std::shared_ptr<QApplication>>>();
+    std::future<std::shared_ptr<QApplication>> application_future =
+        application_promise->get_future();
+    auto visualizer_promise =
+        std::make_shared<std::promise<std::shared_ptr<ThunderbotsVisualizer>>>();
+    std::future<std::shared_ptr<ThunderbotsVisualizer>> visualizer_future =
+        visualizer_promise->get_future();
+    run_visualizer_thread =
+        std::thread(&VisualizerWrapper::createAndRunVisualizer, this, argc, argv,
+                    application_promise, visualizer_promise);
+    application = application_future.get();
+    std::cout << "got application future" << std::endl;
+    visualizer = visualizer_future.get();
```

Comment:
Could we add some comments to this function?

Reply:
done